smithy-swift-codegen/src/test/kotlin/mocks/MockHttpAWSJson11ProtocolGenerator.kt

```diff
@@ -0,0 +1,109 @@
+            .build()
+    }
+}
+class MockAWSJson11HttpProtocolCustomizations() : DefaultHttpProtocolCustomizations()
+class MockHttpAWSJson11ProtocolGenerator : HttpBindingProtocolGenerator() {
+    override val defaultContentType: String = "application/json"
+    override val defaultTimestampFormat: TimestampFormatTrait.Format = TimestampFormatTrait.Format.DATE_TIME
+    override val protocol: ShapeId = AwsJson1_1Trait.ID
+    override fun generateProtocolUnitTests(ctx: ProtocolGenerator.GenerationContext): Int {
+        val requestTestBuilder = HttpProtocolUnitTestRequestGenerator.Builder()
+        val responseTestBuilder = HttpProtocolUnitTestResponseGenerator.Builder()
+        val errorTestBuilder = HttpProtocolUnitTestErrorGenerator.Builder()
+
+        return HttpProtocolTestGenerator(
+            ctx,
+            requestTestBuilder,
+            responseTestBuilder,
+            errorTestBuilder,
+            httpProtocolCustomizable,
+            operationMiddleware,
+            HttpProtocolUnitTestGenerator.SerdeContext("JSONEncoder()", "JSONDecoder()", ".secondsSince1970")
+        ).generateProtocolTests()
+    }
+
+    override val httpProtocolClientGeneratorFactory = TestHttpProtocolClientGeneratorFactory()
+    override val httpProtocolCustomizable = MockAWSJson11HttpProtocolCustomizations()
+    override val codingKeysGenerator: CodingKeysGenerator = DefaultCodingKeysGenerator(CodingKeysCustomizationJsonName())
+    override val httpResponseGenerator: HttpResponseGeneratable = HttpResponseGenerator(
+        unknownServiceErrorSymbol,
+        defaultTimestampFormat,
+        MockHttpResponseBindingErrorGenerator()
+    )
+    override val shouldRenderDecodableBodyStructForInputShapes = true
+    override val shouldRenderCodingKeysForEncodable = true
+
+    override fun renderStructEncode(
+        ctx: ProtocolGenerator.GenerationContext,
+        shapeContainingMembers: Shape,
+        shapeMetadata: Map<ShapeMetadata, Any>,
+        members: List<MemberShape>,
+        writer: SwiftWriter,
+        defaultTimestampFormat: TimestampFormatTrait.Format,
+    ) {
+        val encodeGenerator = StructEncodeGenerator(ctx, members, writer, defaultTimestampFormat)
+        encodeGenerator.render()
+    }
+    override fun renderStructDecode(
+        ctx: ProtocolGenerator.GenerationContext,
+        shapeMetadata: Map<ShapeMetadata, Any>,
+        members: List<MemberShape>,
+        writer: SwiftWriter,
+        defaultTimestampFormat: TimestampFormatTrait.Format,
+    ) {
+        val decodeGenerator = StructDecodeGenerator(ctx, members, writer, defaultTimestampFormat)
+        decodeGenerator.render()
+    }
+
+    override fun addProtocolSpecificMiddleware(ctx: ProtocolGenerator.GenerationContext, operation: OperationShape) {
+        // Intentionally empty
+    }
+
+    override fun getProtocolHttpBindingResolver(ctx: ProtocolGenerator.GenerationContext):
+        HttpBindingResolver = MockJsonHttpBindingResolver(ctx)
+}
```

Comment:
You may need to rebase -- this function i believe needs two parameters.

Reply:
ive rebased like 3x and for some reason still not getting this change but I see what you mean